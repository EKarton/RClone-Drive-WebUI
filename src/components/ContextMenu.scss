.context-menu {
  display: contents;
  cursor: context-menu;
}
