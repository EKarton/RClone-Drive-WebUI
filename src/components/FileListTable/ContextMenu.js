import { Divider, ListItemIcon, ListItemText, Menu, MenuItem } from '@mui/material';
import FileOpenIcon from '@mui/icons-material/FileOpen';
import DownloadIcon from '@mui/icons-material/Download';
import DriveFileMoveIcon from '@mui/icons-material/DriveFileMove';
import ContentCopyIcon from '@mui/icons-material/ContentCopy';
import DriveFileRenameOutlineIcon from '@mui/icons-material/DriveFileRenameOutline';
import DeleteIcon from '@mui/icons-material/Delete';

export default function ContextMenu({
  open,
  onClose,
  menuPosition,
  menuButtonEvents: { onOpen, onDownload, onMove, onCopy, onRename, onDelete },
}) {
  return (
    <Menu
      open={open}
      onClose={onClose}
      anchorReference="anchorPosition"
      anchorPosition={menuPosition}
    >
      <MenuItem onClick={onOpen}>
        <ListItemIcon>
          <FileOpenIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Open</ListItemText>
      </MenuItem>
      <MenuItem onClick={onDownload}>
        <ListItemIcon>
          <DownloadIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Download</ListItemText>
      </MenuItem>
      <Divider />
      <MenuItem onClick={onMove}>
        <ListItemIcon>
          <DriveFileMoveIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Move to</ListItemText>
      </MenuItem>
      <MenuItem onClick={onCopy}>
        <ListItemIcon>
          <ContentCopyIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Copy</ListItemText>
      </MenuItem>
      <MenuItem onClick={onRename}>
        <ListItemIcon>
          <DriveFileRenameOutlineIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Rename</ListItemText>
      </MenuItem>
      <Divider />
      <MenuItem onClick={onDelete}>
        <ListItemIcon>
          <DeleteIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Delete</ListItemText>
      </MenuItem>
    </Menu>
  );
}
