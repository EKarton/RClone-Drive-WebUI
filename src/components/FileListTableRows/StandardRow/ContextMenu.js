import ContentCopyIcon from '@mui/icons-material/ContentCopy';
import DeleteIcon from '@mui/icons-material/Delete';
import DownloadIcon from '@mui/icons-material/Download';
import DriveFileMoveIcon from '@mui/icons-material/DriveFileMove';
import DriveFileRenameOutlineIcon from '@mui/icons-material/DriveFileRenameOutline';
import FileOpenIcon from '@mui/icons-material/FileOpen';
import Divider from '@mui/material/Divider';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import PropTypes from 'prop-types';

export default function ContextMenu({
  open,
  onClose,
  menuPosition,
  menuButtonEvents: { onOpen, onDownload, onMove, onCopy, onRename, onDelete },
}) {
  return (
    <Menu
      open={open}
      onClose={onClose}
      anchorReference="anchorPosition"
      anchorPosition={menuPosition}
      data-testid="filetable-context-menu"
    >
      <MenuItem onClick={onOpen} data-testid="open">
        <ListItemIcon>
          <FileOpenIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Open</ListItemText>
      </MenuItem>
      <MenuItem onClick={onDownload} data-testid="download">
        <ListItemIcon>
          <DownloadIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Download</ListItemText>
      </MenuItem>
      <Divider />
      <MenuItem onClick={onMove} data-testid="move">
        <ListItemIcon>
          <DriveFileMoveIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Move to</ListItemText>
      </MenuItem>
      <MenuItem onClick={onCopy} data-testid="copy">
        <ListItemIcon>
          <ContentCopyIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Copy</ListItemText>
      </MenuItem>
      <MenuItem onClick={onRename} data-testid="rename">
        <ListItemIcon>
          <DriveFileRenameOutlineIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Rename</ListItemText>
      </MenuItem>
      <Divider />
      <MenuItem onClick={onDelete} data-testid="delete">
        <ListItemIcon>
          <DeleteIcon fontSize="small" />
        </ListItemIcon>
        <ListItemText>Delete</ListItemText>
      </MenuItem>
    </Menu>
  );
}

ContextMenu.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
  menuPosition: PropTypes.shape({
    left: PropTypes.number,
    top: PropTypes.number,
  }),
  menuButtonEvents: PropTypes.shape({
    onOpen: PropTypes.func,
    onDownload: PropTypes.func,
    onMove: PropTypes.func,
    onCopy: PropTypes.func,
    onRename: PropTypes.func,
    onDelete: PropTypes.func,
  }),
};
