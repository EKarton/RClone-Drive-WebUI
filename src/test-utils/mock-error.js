/**
 * Replaces the error object's stack trace with mock data
 * @param {Error} error the Error object
 * @returns {Error} the same error, but with the stack trace mocked
 */
export function mockErrorStackTrace(error) {
  error.stack = 'Mock content for stack trace';
  return error;
}
