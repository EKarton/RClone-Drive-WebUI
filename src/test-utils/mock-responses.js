export const mockFiles = {
  list: [
    {
      Path: '.rclone',
      Name: '.rclone',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-11-01T21:09:05.237Z',
      IsDir: true,
      ID: '1a4mI4vsGKypqFFPwLnrFid1ZdvlhirDi',
    },
    {
      Path: 'Documents',
      Name: 'Documents',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T01:51:05.982Z',
      IsDir: true,
      ID: '1_za5Kqy9ZjXZ7rsztGpKhtJLTuNCKv8q',
    },
    {
      Path: 'Games',
      Name: 'Games',
      Size: -9,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T05:14:59.237Z',
      IsDir: true,
      ID: '1j6mAjxQoJokiU8yWJYgyWYIiSIgH1G7M',
    },
    {
      Path: 'Literature',
      Name: 'Literature',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-29T07:10:05.059Z',
      IsDir: true,
      ID: '1dj1T5YkjRQZNiQ5efCA8PzideJb88d3K',
    },
    {
      Path: 'Pictures',
      Name: 'Pictures',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:39:11.850Z',
      IsDir: true,
      ID: '1D2PpmoIT23NqyTnolmohZ0UxvOd2D7n4',
    },
    {
      Path: 'Software',
      Name: 'Software',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:45:02.471Z',
      IsDir: true,
      ID: '1zzr0Bwp9tel66rCUKz8mZIXRBR2kuywr',
    },
    {
      Path: 'Videos',
      Name: 'Videos',
      Size: -8,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T04:22:24.997Z',
      IsDir: true,
      ID: '1eJlJ1PT7b9XY3Q5tLeU49BNkug259986',
    },
    {
      Path: 'backup.sh',
      Name: 'backup.sh',
      Size: 1110,
      MimeType: 'application/x-sh',
      ModTime: '2021-11-14T18:58:44.655Z',
      IsDir: false,
    },
  ],
};

export const mockRemotes = {
  remotes: ['googledrive', 'onedrive', 'icloud'],
};

export const mockPictures = {
  list: [
    {
      Path: 'Pictures/2012/20120517_171428.JPG',
      Name: '20120517_171428.JPG',
      Size: 335408,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:14:30.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_135206.JPG',
      Name: '20120517_135206.JPG',
      Size: 262719,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T17:52:06.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_204557.JPG',
      Name: '20120516_204557.JPG',
      Size: 245109,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T00:45:58.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_174510.JPG',
      Name: '20120517_174510.JPG',
      Size: 271914,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:45:12.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120820_144833.JPG',
      Name: '20120820_144833.JPG',
      Size: 2662880,
      MimeType: 'image/jpeg',
      ModTime: '2012-08-20T18:48:32.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_094409.JPG',
      Name: '20120517_094409.JPG',
      Size: 280596,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T13:44:10.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181954.JPG',
      Name: '20120516_181954.JPG',
      Size: 416424,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:19:54.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181809.JPG',
      Name: '20120516_181809.JPG',
      Size: 211575,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:18:10.000Z',
      IsDir: false,
    },
  ],
};
