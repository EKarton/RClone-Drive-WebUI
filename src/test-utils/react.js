import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { ColorModeProvider } from 'contexts/ColorMode/index';
import { JobQueueProvider } from 'contexts/JobQueue/index';
import { JobsListDialogProvider } from 'contexts/JobsListDialog/index';
import { RCloneInfoProvider } from 'contexts/RCloneInfo';
import { RecentPicturesProvider } from 'contexts/RecentPicturesList';
import { SnackbarProvider } from 'notistack';

const customRender = (
  component,
  { initialRCloneInfoState, initialRecentPicturesState, initialJobQueueState } = {},
  { route = '/' } = {}
) => {
  const renderedComponent = render(
    <MemoryRouter initialEntries={[route]}>
      <RCloneInfoProvider defaultState={initialRCloneInfoState}>
        <RecentPicturesProvider defaultState={initialRecentPicturesState}>
          <ColorModeProvider>
            <JobQueueProvider defaultState={initialJobQueueState}>
              <JobsListDialogProvider>
                <SnackbarProvider
                  maxSnack={3}
                  anchorOrigin={{ horizontal: 'right', vertical: 'top' }}
                  autoHideDuration={3000}
                >
                  {component}
                </SnackbarProvider>
              </JobsListDialogProvider>
            </JobQueueProvider>
          </ColorModeProvider>
        </RecentPicturesProvider>
      </RCloneInfoProvider>
    </MemoryRouter>
  );

  return {
    ...renderedComponent,
  };
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent, render };
