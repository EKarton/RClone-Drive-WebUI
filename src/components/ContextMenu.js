import { Divider, ListItemIcon, ListItemText, Menu, MenuItem } from '@mui/material';
import FileOpenIcon from '@mui/icons-material/FileOpen';
import DownloadIcon from '@mui/icons-material/Download';
import DriveFileMoveIcon from '@mui/icons-material/DriveFileMove';
import ContentCopyIcon from '@mui/icons-material/ContentCopy';
import DriveFileRenameOutlineIcon from '@mui/icons-material/DriveFileRenameOutline';
import DeleteIcon from '@mui/icons-material/Delete';

import { useState } from 'react';
import './ContextMenu.scss';

export default function ContextMenu(props) {
  const { onOpen, onDownload, onDelete, onMove, onCopy, onRename } = props;
  const { onMenuChanged } = props;
  const { children } = props;

  const [contextMenu, setContextMenu] = useState(null);

  const handleContextMenuClicked = (func) => () => {
    setContextMenu(null);
    onMenuChanged({ isOpen: false });
    func();
  };

  const handleContextMenu = (e) => {
    e.preventDefault();

    const menuPosition = { left: e.clientX - 2, top: e.clientY - 4 };
    const newContextMenu = contextMenu === null ? menuPosition : null;
    setContextMenu(newContextMenu);
    onMenuChanged({ isOpen: newContextMenu !== null });
  };

  const handleClose = () => {
    onMenuChanged({ isOpen: false });
    setContextMenu(null);
  };

  return (
    <div onContextMenu={handleContextMenu} className="context-menu">
      {children}
      <Menu
        open={contextMenu !== null}
        onClose={handleClose}
        anchorReference="anchorPosition"
        anchorPosition={contextMenu !== null ? contextMenu : undefined}
      >
        <MenuItem onClick={handleContextMenuClicked(onOpen)}>
          <ListItemIcon>
            <FileOpenIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText>Open</ListItemText>
        </MenuItem>
        <MenuItem onClick={handleContextMenuClicked(onDownload)}>
          <ListItemIcon>
            <DownloadIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText>Download</ListItemText>
        </MenuItem>
        <Divider />
        <MenuItem onClick={handleContextMenuClicked(onMove)}>
          <ListItemIcon>
            <DriveFileMoveIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText>Move to</ListItemText>
        </MenuItem>
        <MenuItem onClick={handleContextMenuClicked(onCopy)}>
          <ListItemIcon>
            <ContentCopyIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText>Copy</ListItemText>
        </MenuItem>
        <MenuItem onClick={handleContextMenuClicked(onRename)}>
          <ListItemIcon>
            <DriveFileRenameOutlineIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText>Rename</ListItemText>
        </MenuItem>
        <Divider />
        <MenuItem onClick={handleContextMenuClicked(onDelete)}>
          <ListItemIcon>
            <DeleteIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText>Delete</ListItemText>
        </MenuItem>
      </Menu>
    </div>
  );
}

ContextMenu.defaultProps = {
  onOpen: () => {},
  onDownload: () => {},
  onDelete: () => {},
  onMove: () => {},
  onCopy: () => {},
  onRename: () => {},
};
