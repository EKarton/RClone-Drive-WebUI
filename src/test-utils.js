import { FileViewerProvider, initialState } from 'contexts/FileViewerStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState = initialState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  return render(
    <FileViewerProvider defaultState={initialFileViewerState}>
      <Router history={history}>{component}</Router>
    </FileViewerProvider>
  );
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
