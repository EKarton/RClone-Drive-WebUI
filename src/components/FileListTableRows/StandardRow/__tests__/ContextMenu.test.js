import { render } from 'test-utils/react';
import ContextMenu from '../ContextMenu';

describe('ContextMenu', () => {
  it('should match snapshot given it is open', () => {
    const { baseElement } = render(
      <ContextMenu
        open
        onClose={jest.fn()}
        menuPosition={{ left: 200, top: 300 }}
        menuButtonEvents={{
          onOpen: jest.fn(),
          onDownload: jest.fn(),
          onMove: jest.fn(),
          onCopy: jest.fn(),
          onRename: jest.fn(),
          onDelete: jest.fn(),
        }}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
