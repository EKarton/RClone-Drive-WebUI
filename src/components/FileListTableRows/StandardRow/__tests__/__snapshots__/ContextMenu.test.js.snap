// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu should match snapshot given it is open 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiPopover-root MuiMenu-root MuiModal-root css-10nakn3-MuiModal-root-MuiPopover-root-MuiMenu-root"
    data-testid="filetable-context-menu"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-1hqw972-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper MuiMenu-paper css-1xomo8h-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
      style="--Paper-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12); opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 300px; left: 200px; transform-origin: 0px 0px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiList-padding MuiMenu-list css-99wn1p-MuiList-root-MuiMenu-list"
        role="menu"
        tabindex="-1"
      >
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1h4bzp1-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="open"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="FileOpenIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 2H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H15v-8h5V8zm-1 7V3.5L18.5 9zm4 12.66V16h5.66v2h-2.24l2.95 2.95-1.41 1.41L19 19.41v2.24h-2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Open
            </span>
          </div>
        </li>
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1h4bzp1-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="download"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="DownloadIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 20h14v-2H5zM19 9h-4V3H9v6H5l7 7z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Download
            </span>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1h4bzp1-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="move"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="DriveFileMoveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 6h-8l-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2m-6 12v-3h-4v-4h4V8l5 5z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Move to
            </span>
          </div>
        </li>
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1h4bzp1-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="copy"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="ContentCopyIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2m0 16H8V7h11z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Copy
            </span>
          </div>
        </li>
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1h4bzp1-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="rename"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="DriveFileRenameOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18.41 5.8 17.2 4.59c-.78-.78-2.05-.78-2.83 0l-2.68 2.68L3 15.96V20h4.04l8.74-8.74 2.63-2.63c.79-.78.79-2.05 0-2.83M6.21 18H5v-1.21l8.66-8.66 1.21 1.21zM11 20l4-4h6v4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Rename
            </span>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1h4bzp1-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="delete"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Delete
            </span>
          </div>
        </li>
      </ul>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
